gui/src/app/state/spend/step.rs

```diff
@@ -175,110 +188,121 @@ impl DefineSpend {
             }
         }
     }
-    fn auto_select_coins(&mut self, daemon: Arc<dyn Daemon + Sync + Send>) {
-        // Set non-input values in the same way as for user selection.
-        let mut outputs: HashMap<Address<address::NetworkUnchecked>, u64> = HashMap::new();
-        for recipient in &self.recipients {
-            outputs.insert(
-                Address::from_str(&recipient.address.value).expect("Checked before"),
-                recipient.amount().expect("Checked before"),
-            );
+    /// redraft calculates the amount left to select and auto selects coins
+    /// if the user did not select a coin manually
+    fn redraft(&mut self, daemon: Arc<dyn Daemon + Sync + Send>) {
+        if !self.form_values_are_valid() || self.recipients.is_empty() {
```

Comment:
Perhaps in a follow-up we could also use this function for a self-transfer to check if the user sets a feerate that is too high and will not give any change.

Reply:
Yes, we could also use this function to calculate the amount a `max` button feature.

Reply:
Opened https://github.com/wizardsardine/liana/issues/868 to track this.